packages/cli/src/ui/themes/theme-manager.ts

```diff
@@ -65,44 +67,156 @@ class ThemeManager {
    */
   loadCustomThemes(customThemesSettings?: Record<string, CustomTheme>): void {
     this.customThemes.clear();
+    this.resolvedNameToTheme.clear(); // Clear this map as well
+    this.themeFilePaths.clear();
 
     if (!customThemesSettings) {
       return;
     }
 
-    for (const [name, customThemeConfig] of Object.entries(
+    for (const [key, customThemeConfig] of Object.entries(
       customThemesSettings,
     )) {
+      // Check if it's a file-based theme
+      if (customThemeConfig.path) {
+        this.loadThemeFromFileWithKey(
```

Comment:
bonus points: fail if there are are any other properties. if there is a path there should not be any other entries.